test/e2e/apimachinery/crd_publish_openapi.go

```diff
@@ -414,7 +415,7 @@ func waitForDefinition(c k8sclientset.Interface, name string, schema []byte) err
 		// drop properties and extension that we added
 		dropDefaults(&d)
 		if !apiequality.Semantic.DeepEqual(expect, d) {
-			return false, fmt.Sprintf("spec.SwaggerProps.Definitions[\"%s\"] not match; expect: %v, actual: %v", name, expect, d)
+			return false, fmt.Sprintf("spec.SwaggerProps.Definitions[\"%s\"] not match;\n%s", name, diff.ObjectReflectDiff(expect, d))
```

Comment:
does this really work? we have private fields in the `$ref` field if I remember right. The reflect diff complains that it cannot read those. So better double check.

Reply:
hmm, not sure... just trying to get a more useful message than:

> failed to wait for OpenAPI spec validating condition: timed out waiting for the condition; lastMsg: spec.SwaggerProps.Definitions["io.k8s.crd-publish-openapi-test-common-group.v6.E2e-test-crd-publish-openapi-6170-crd"] not match; expect: {{map[]} { {{<nil> {[]} false false false false false}}  Waldo CRD for Testing [object]   <nil> <nil> false <nil> false <nil> <nil>  <nil> <nil> false <nil> [] <nil> <nil> [] <nil> [] [] [] <nil> map[spec:{{map[]} { {{<nil> {[]} false false false false false}}  Specification of Waldo [object]   <nil> <nil> false <nil> false <nil> <nil>  <nil> <nil> false <nil> [] <nil> <nil> [] <nil> [] [] [] <nil> map[dummy:{{map[]} { {{<nil> {[]} false false false false false}}  Dummy property. []   <nil> <nil> false <nil> false <nil> <nil>  <nil> <nil> false <nil> [] <nil> <nil> [] <nil> [] [] [] <nil> map[] <nil> map[] map[] <nil> map[]} { false <nil> <nil> <nil>} map[]}] <nil> map[] map[] <nil> map[]} { false <nil> <nil> <nil>} map[]} status:{{map[]} { {{<nil> {[]} false false false false false}}  Status of Waldo [object]   <nil> <nil> false <nil> false <nil> <nil>  <nil> <nil> false <nil> [] <nil> <nil> [] <nil> [] [] [] <nil> map[bars:{{map[]} { {{<nil> {[]} false false false false false}}  List of Bars and their statuses. []   <nil> <nil> false <nil> false <nil> <nil>  <nil> <nil> false <nil> [] <nil> <nil> [] <nil> [] [] [] <nil> map[] <nil> map[] map[] <nil> map[]} { false <nil> <nil> <nil>} map[]}] <nil> map[] map[] <nil> map[]} { false <nil> <nil> <nil>} map[]}] <nil> map[] map[] <nil> map[]} { false <nil> <nil> <nil>} map[]}, actual: {{map[]} { {{<nil> {[]} false false false false false}}   [object]   <nil> <nil> false <nil> false <nil> <nil>  <nil> <nil> false <nil> [] <nil> <nil> [] <nil> [] [] [] <nil> map[] <nil> map[] map[] <nil> map[]} { false <nil> <nil> <nil>} map[]}